packages/core/upload/admin/src/hooks/useAssets.js

```diff
@@ -71,6 +71,31 @@ export const useAssets = ({ skipWhen = false, query = {} } = {}) => {
     enabled: !skipWhen,
     staleTime: 0,
     cacheTime: 0,
+    select(data) {
+      if (data?.results && Array.isArray(data.results)) {
+        return {
+          ...data,
+          results: data.results
+            /**
+             * Filter out assets that don't have a name.
+             * So we don't try to render them as assets
+             * and get errors.
+             */
+            .filter((asset) => asset.name)
+            .map((asset) => ({
+              ...asset,
+              /**
+               * Mime and ext cannot be null in the front-end because
+               * we expect them to be strings and use the `includes` method.
+               */
+              mime: asset.mime ?? '',
```

Comment:
What do you think about setting the fallback to `application/octet-stream` instead (which is the most generic mime-type possible)?

Reply:
sounds good to me , we could check if we can not get the mime type of the asset and assign `application/octet-stream` by default! good point

Reply:
> What do you think about setting the fallback to application/octet-stream instead (which is the most generic mime-type possible)?

I think one thing to think about is how this would impact the `allowedTypes` filtering? 🤔 is that based on `mime`?

Reply:
@joshuaellis Good point it is and it is not very smart: it only uses the first part of the mime-type, e.g. the filter for image searches all assets with a mime-type starting with image, such as image/png.

But it wouldn't break the file-filter, because this is the filter it creates: `filters[$and][0][mime][$not][$contains][0]=image&filters[$and][0][mime][$not][$contains][1]=video` in the FE.

Reply:
Me and @Marc-Roig also have a session today to look at some "broken assets" and understand why it's failing in the first place. Because the mime comes from the file.type so it's odd that it breaks.